tests/checkpoint/test_checkpoint.py

```diff
@@ -851,18 +851,181 @@ def test_newstyle_checkpoint_config_substitution_simple(
+                "action": {
+                    "class_name": "UpdateDataDocsAction",
+                },
+            },
+            {
+                "name": "new_action_from_template_2",
+                "action": {"class_name": "Template2SpecialAction"},
+            },
+        ],
+        evaluation_parameters={
+            "environment": "$GE_ENVIRONMENT",
+            "tolerance": 1.0e-2,
+            "aux_param_0": "$MY_PARAM",
+            "aux_param_1": "1 + $MY_PARAM",
+        },
+        runtime_configuration={
+            "result_format": "BASIC",
+            "partial_unexpected_count": 20,
+        },
+    )
+    nested_checkpoint_template_config_2_key = ConfigurationIdentifier(
+        configuration_key=nested_checkpoint_template_config_2.name
     )
     context.checkpoint_store.set(
-        key=checkpoint_template_config_key, value=checkpoint_template_config
+        key=nested_checkpoint_template_config_2_key,
+        value=nested_checkpoint_template_config_2,
     )
 
-    # add child config
+    nested_checkpoint_template_config_3 = CheckpointConfig(
+        config_version=1,
+        name="my_nested_checkpoint_template_3",
+        template_name="my_nested_checkpoint_template_2",
+        run_name_template="%Y-%M-foo-bar-template-$VAR-template-3",
+        action_list=[
+            {
+                "name": "store_validation_result",
+                "action": {
+                    "class_name": "StoreValidationResultAction",
+                },
+            },
+            {
+                "name": "store_evaluation_params",
+                "action": {
+                    "class_name": "MyCustomStoreEvaluationParametersActionTemplate3",
+                },
+            },
+            {
+                "name": "update_data_docs",
+                "action": {
+                    "class_name": "UpdateDataDocsAction",
+                },
+            },
+            {
+                "name": "new_action_from_template_3",
+                "action": {"class_name": "Template3SpecialAction"},
+            },
+        ],
+        evaluation_parameters={
+            "environment": "$GE_ENVIRONMENT",
+            "tolerance": 1.0e-2,
+            "aux_param_0": "$MY_PARAM",
+            "aux_param_1": "1 + $MY_PARAM",
+            "template_3_key": 123,
+        },
+        runtime_configuration={
+            "result_format": "BASIC",
+            "partial_unexpected_count": 20,
+            "template_3_key": "bloopy!",
+        },
+    )
+    nested_checkpoint_template_config_3_key = ConfigurationIdentifier(
+        configuration_key=nested_checkpoint_template_config_3.name
+    )
+    context.checkpoint_store.set(
+        key=nested_checkpoint_template_config_3_key,
+        value=nested_checkpoint_template_config_3,
+    )
+
+    return context
+
+
+def test_newstyle_checkpoint_config_substitution_simple(
+    context_with_checkpoint_templates,
+):
+    context = context_with_checkpoint_templates
+
```

Comment:
I am not sure whether or not this fixture is needed.  I added a bunch of tests, including those for the nested Checkpoint YML configuration (above this fixture).  Maybe all we need to do is uncommented the variable substitution, remove comments containing "ALEX", and make sure that those tests pass?  Just wondering.  Thanks.

Reply:
Okay - makes total sense, since I borrowed a lot from a test you created before, before there was an actual fixture. I think I maybe have added a few things, but if their functionally the same, I agree we should just have one and put it in conftest. Will examine.

Reply:
- After going over the tests and fixtures, I think we should keep the ones I added since they also test running checkpoints with runtime kwargs, as well as more deeply nested configs
- I moved the fixture I added with all the checkpoint templates into conftest and renamed it to be more clear

Per conversation I had with @jcampbell, I'm holding off on eval param processing for now, since this is a new feature that deserves its own design review. Therefore, I've left your "Alex" comments relating to the eval params keys. I did however uncomment the parts that were purely var sub related (run_name key) and updated the test, which is passing.